### PR TITLE
feat(datasource): add demo datasets to apps

### PR DESCRIPTION
**Pull request by slettner**

### CODE REVIEW

**Comment by slettner** [comment]
This PR only makes apps list their available demo datasets.
This is needed to list them in the storefront. 

The `get_data` logic is not used yet. 
All the data loading logic lives inside `data_loading` module. Instead, apps should know all about the data they support and use the `Datasources` to implement loading logic. `data_loading` should be replaced by `data_sources` in the future. 

For example, if a user specifies a custom docarray via the storefront, the now library will construct the data by creating a `CustomDocarray(Datasource)` and call its `get_data` function.

**Comment by slettner** [comment]
In think datasource can be grouped into 
- DemoDataset
- CustomDocarray
- LocalFiles
- Database

File: now/datasource/datasource.py [hunk 1]
```diff
@@ -1,46 +1,61 @@
...
-        #  This signature misses the mapping arg from the baseclass.
-        #  The baseclass interface is not really guiding the user a lot at the moment.
-        #  Maybe we should start with an explicit ObjectStorageDataset or something like that
-        #  because all our app are using that atm. and abstract the interface furhter once we work on
-        #  the database table column mapping stuff?
-
-        url = f'{BASE_STORAGE_URL}/{self.modality}/{self.ds_id}{("." + quality) if quality is not None else ""}.bin'
-        # TODO  return document array from url
-        raise NotImplementedError()
+        self.modalities_folder = modality_folder
```
Comment (makram93) [inline now/datasource/datasource.py:43]: What is this needed for?

**Review by joschkabraun — APPROVED** [review]
LGTM!

**Pull request merged**